### ISSUE TITLE
Support Math Equations and Images

### ISSUE THREAD

**Issue by athul** [hackstream/zettel#10]
Hey, 
I was trying to build a wiki of my own and found this in the FossHack Chat Group. I was intrigued and wanted to try it. I write math equations(logical equations) in KaTeX and wanted to have a self-hostable solution for the same. 

I found a Bug and Implemented a feature(will send a PR:wink:).
The Bug I found was that zettle doesn't handle images well. I got a full panic when there was an image in the `contents` directory. 

Here is the panic op

```
❯./zettel.bin build         
DEBU[2020-09-15T16:10:10+05:30] verbose logging enabled                      
INFO[2020-09-15T16:10:10+05:30] Starting zettel...                           
2020/09/15 16:10:10 reading config: zettel.toml
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/hackstream/zettel/internal/pipeline.ReadFiles.func1(0xc0003809a0, 0x18, 0x175f4c0, 0xc000113a00, 0x0, 0x0, 0x10, 0xc0002bf4a8)
        /Users/athul/go/src/github.com/athul/zettel/internal/pipeline/pipeline.go:45 +0x58e
path/filepath.walk(0xc0003809a0, 0x18, 0x175f4c0, 0xc000113a00, 0xc0002bf658, 0x0, 0x0)
        /usr/local/opt/go/libexec/src/path/filepath/path.go:360 +0x425
path/filepath.walk(0x16539f9, 0x7, 0x175f4c0, 0xc000113380, 0xc0002bf658, 0x0, 0xc000205c70)
        /usr/local/opt/go/libexec/src/path/filepath/path.go:384 +0x2ff
path/filepath.Walk(0x16539f9, 0x7, 0xc0002bf658, 0x26, 0x10)
        /usr/local/opt/go/libexec/src/path/filepath/path.go:406 +0xff
github.com/hackstream/zettel/internal/pipeline.ReadFiles(0x16539f9, 0x7, 0x0, 0x0, 0x1766720, 0x159d8c0, 0x1583c02)
        /Users/athul/go/src/github.com/athul/zettel/internal/pipeline/pipeline.go:24 +0x72
main.(*Hub).build(0xc00008f310, 0xc00021ee00, 0x164f0e4, 0x1)
        /Users/athul/go/src/github.com/athul/zettel/cmd/zettel/build.go:78 +0x4b
main.(*Hub).MustHaveConfig.func1(0xc00021ee00, 0x1, 0x1)
        /Users/athul/go/src/github.com/athul/zettel/cmd/zettel/middleware.go:19 +0xe6
github.com/urfave/cli/v2.(*Command).Run(0xc0001dbe60, 0xc00021ed40, 0x0, 0x0)
        /Users/athul/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/command.go:164 +0x4e0
github.com/urfave/cli/v2.(*App).RunContext(0xc000082480, 0x175cbe0, 0xc000166ce0, 0xc0000aa000, 0x2, 0x2, 0x0, 0x0)
        /Users/athul/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/app.go:306 +0x814
github.com/urfave/cli/v2.(*App).Run(...)
        /Users/athul/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/app.go:215
main.main()
        /Users/athul/go/src/github.com/athul/zettel/cmd/zettel/main.go:87 +0x4cc
```

I implemented the math support with a [goldmark extension](https://github.com/litao91/goldmark-mathjax). Specifically a [personal fork of the extension](https://github.com/athul/goldmark-mathjax) due to a compilation error in the original extension. 

Here is an image without the math support

![Screen Shot 2020-09-15 at 4 13 50 PM](https://user-images.githubusercontent.com/40897573/93201180-746d3a00-f76e-11ea-8596-1bd2b31bf8c8.png)

Here is an image with Math Support from KaTex
![Screen Shot 2020-09-15 at 4 18 33 PM](https://user-images.githubusercontent.com/40897573/93201656-1bea6c80-f76f-11ea-9759-a37c6e312649.png)

This Math support is made possible by KaTeX Js and CSS in the Head tag of `header.tmpl`

**Comment by iamd3vil** [comment]
Thanks for reporting the bug @athul. I found about this today myself. I am working on a fix today.

The Math equations support sound cool. Feel free to send a PR regarding that.

**Comment by iamd3vil** [comment]
I have released `v0.4.0` just now which should fix the bug. Closing this issue. 

Please try it out and feel free to reopen in case it doesn't fix it.
